### DISCUSSION TITLE
gl_Vertex and gl_Normal?

### DISCUSSION THREAD

**Discussion by aVersionOfReality** [bnpr/Malt#258] [Q&A] [answered]
I'm trying to use some code snippets for Vertex shaders that use these, but Malt gives me: "error C7616: global variable gl_Vertex is removed after version 140."

What is the proper substitute for these? Is it the V.position and V.normal from the Common Vertex Shader Input?

**Comment by pragma37** [comment, marked as answer]
I would be wary with those code snippets, gl_Vertex has been deprecated for almost 2 decades.
I even had to google it because I wasn't even sure what it was used for.

From nodes, position and normal can always be read from *surface position* and *surface normal*.
If you want to override the vertex shader outputs, use the Vertex Shader Output.
